packages/core/src/browser/opener-service.ts

```diff
@@ -84,12 +92,27 @@ export async function open(openerService: OpenerService, uri: URI, options?: Ope
 
 @injectable()
 export class DefaultOpenerService implements OpenerService {
+    // Collection of open-handlers for custom-editor contributions.
+    protected readonly customEditorOpenHandlers: OpenHandler[] = [];
+
+    protected readonly onOpenersStateChangedEmitter = new Emitter<void>();
+    readonly onOpenersStateChanged = this.onOpenersStateChangedEmitter.event;
 
     constructor(
         @inject(ContributionProvider) @named(OpenHandler)
         protected readonly handlersProvider: ContributionProvider<OpenHandler>
     ) { }
 
+    addHandler(openHandler: OpenHandler): Disposable {
+        this.customEditorOpenHandlers.push(openHandler);
+        this.onOpenersStateChangedEmitter.fire();
+
+        return Disposable.create(() => {
+            this.customEditorOpenHandlers.splice(this.customEditorOpenHandlers.indexOf(openHandler), 1);
+            this.onOpenersStateChangedEmitter.fire();
+        });
+    }
```

Comment:
minor: with the newly added API, and the fact we can add/remove handlers dynamically it may be worthwhile to test it through the `opener-service.spec.ts` although I will not hold back the pull-request for it :+1:

Reply:
added "addHandler" test to opener-service.spec.ts. Thanks